packages/docusaurus-plugin-content-blog/src/options.ts

```diff
@@ -129,9 +130,10 @@ const PluginOptionSchema = Joi.object<PluginOptions>({
   }).default(DEFAULT_OPTIONS.feedOptions),
   authorsMapPath: Joi.string().default(DEFAULT_OPTIONS.authorsMapPath),
   readingTime: Joi.function().default(() => DEFAULT_OPTIONS.readingTime),
-  sortPosts: Joi.string()
-    .valid('descending', 'ascending')
+  sortPosts: Joi.alternatives()
+    .try(Joi.string().valid('descending', 'ascending'))
     .default(DEFAULT_OPTIONS.sortPosts),
+  processPosts: Joi.function().optional().default(DEFAULT_OPTIONS.processPosts),
```

Comment:
```suggestion
  processPosts: Joi.function().optional().default(() => DEFAULT_OPTIONS.processPosts),
```

Joi quirk: it executes the function to get the default value